web/src/components/ReaderLayout/ReaderLayout.tsx

```diff
@@ -85,7 +86,8 @@ const ReaderLayout: React.FC<{
    */
   const getProxiedResource = (proxyUrl?: string) => async (href: string) => {
     // Generate the resource URL using the proxy if the URI is not stored in S3
-    const url: string = proxyUrl && !href.includes('drb-files')
+    setProxyResource(proxyUrl && !href.includes('drb-files'));
```

Comment:
Should we move this to the point where we first receive the manifest data so that it can handle multi-PDF resources as well?

Reply:
e.g.

```ts
const containsSelfHosted = manifest?.readingOrder?.some((link) =>
  link.href.includes("drb-files")
);
setProxyResource(proxyUrl && !isLimitedAccess && !containsSelfHosted);
```

Reply:
lemme see! good callout

Reply:
I think we are gonna keep the changes up in the getProxiedResource function - mainly because there could be resources that are self hosted but those that are not that are part of the same record - unlikely but something to consider